src/file_patcher.rs

```diff
@@ -1,107 +1,86 @@
 use anyhow::{Context, Result};
 use colored::*;
-use difference::{Changeset, Difference};
+
+use std::collections::BTreeMap;
 use std::fs::File;
 use std::io::{BufRead, BufReader};
 use std::path::{Path, PathBuf};
 
-use crate::line_patcher::LinePatcher;
+use crate::line_patcher::{self, LineReplacements};
 use crate::query::Query;
 
 pub struct FilePatcher {
-    replacements: Vec<Replacement>,
+    // Note: BTreeMap because order matters here
+    replacements: BTreeMap<usize, LineReplacements>,
```

Comment:
not sure why I introduced a LineReplacement struct here in a map, instead of modifying the exitisting Replacement struct and keeping it a vec